### PR TITLE
Improved FunctionList's NSIS parser

### PR DESCRIPTION
**Pull request by Negrutiu**
Improved FunctionList's **NSIS syntax:**
- Added **!echo** keyword
- Detect functions with comments following the function name:
 ```
 Function Test1  ; Comment1
  FunctionEnd

  Function Test2  /* Comment2 */
  FunctionEnd
```

### CODE REVIEW

File: PowerEditor/src/functionList.xml [hunk 1]
```diff
@@ -318,13 +318,13 @@
 			<parser
 				id         ="nsis_syntax"
 				displayName="NSIS Syntax"
-				commentExpr="(?s:/\*.*?\*/)|(?m-s:[#;].*?$)"
+				commentExpr="(?s:^[\t\x20]*/\*.*?\*/)|(?m-s:^[\t\x20]*[#;].*?$)"
```
Comment (MAPJe71) [inline PowerEditor/src/functionList.xml]: The at-start-of-line (`^`) and optional leading blanks (`[\t\x20]*`) don't improve the comment detection but rather make it worse i.e. it prevents detecting trailing comment.

File: PowerEditor/src/functionList.xml [hunk 2]
```diff
@@ -318,13 +318,13 @@
...
 			>
 				<function
-					mainExpr="^[\t\x20]*(!macro|Function|Section|SectionGroup)[\t\x20]+[^\r\n]*$"
+					mainExpr="^[\t\x20]*(!macro|Function|Section|SectionGroup|!echo)[\t\x20]+[^\r\n]*$"
```
Comment (MAPJe71) [inline PowerEditor/src/functionList.xml]: The `!echo` will list text messages in the FunctionList tree. 
That might be your personal preference but is not something to include in the list of official parsers.

File: PowerEditor/src/functionList.xml (hunk 1 above)
Comment (Negrutiu) [inline PowerEditor/src/functionList.xml]: Here's a perfectly valid NSIS script: [Test.nsi.txt](https://github.com/notepad-plus-plus/notepad-plus-plus/files/545170/Test.nsi.txt)
No function is identified with the default parser rules.

With the changes I've suggested (almost) all functions are detected. SubTest4 is of course incorrectly displayed, but still, it becomes much easier to navigate inside large scripts.

Nevertheless, not a critical problem. I'm waiting for the improved parsers...

File: PowerEditor/src/functionList.xml (hunk 2 above)
Comment (Negrutiu) [inline PowerEditor/src/functionList.xml]: That sounds great

**Pull request closed without merging**